src/Type/UnionType.php

```diff
@@ -45,6 +51,12 @@ class UnionType implements CompoundType
 
 	use NonGeneralizableTypeTrait;
 
+	public const EQUAL_UNION_CLASSES = [
+		DateTimeInterface::class => [DateTimeImmutable::class, DateTime::class],
+		Throwable::class => [Error::class, Exception::class], // phpcs:ignore SlevomatCodingStandard.Exceptions.ReferenceThrowableOnly.ReferencedGeneralException
+		Traversable::class => [IteratorAggregate::class, Iterator::class],
```

Comment:
I'd accept part of this PR - the refactoring into this constant. Without addition of the Traversable entry.